src/distilabel/pipeline.py

```diff
@@ -425,13 +425,13 @@ def _build_dataset(  # noqa: C901
                 "raw_generation_responses",
             ] + self.generator.task.output_args_names
 
-            if len(generations) < len(dataset):
-                generations.extend(
-                    [
-                        {key: None for key in generator_column_names}
-                        for _ in range(len(dataset) - len(generations))
-                    ]
-                )
+            # if len(generations) < len(dataset):
+            #     generations.extend(
+            #         [
+            #             {key: None for key in generator_column_names}
+            #             for _ in range(len(dataset) - len(generations))
+            #         ]
+            #     )
```

Comment:
Do we need this data commented?

Reply:
I forgot about that. I'm not sure of the use of that lines, but the code fails with that code. If that code is activated the dataset isn't correctly generated, it would only save the dataset up to a given batch, loosing that part of the dataset already generated. Commenting that part, everything seems to work fine.
@alvarobartt, @gabrielmbmb, could you help me here to know what this piece of code is for?

Reply:
This is for a very unlikely (I think it's not possible anymore for it to happen) case in which we haven't generated texts for all the rows, but we can remove this block because the solution is not good either.

Reply:
In that case erased as it never happened